File: Deepseek-Coder-33b-Instruct/stage2/acc_map_data.f90
#ifndef T1
!T1:acc map data,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER, PARAMETER :: N = 100
        REAL(KIND=4), DIMENSION(:), ALLOCATABLE :: host_data
        REAL(KIND=4), DIMENSION(:), ALLOCATABLE :: device_data
        INTEGER :: i

        ! Allocate host memory
        ALLOCATE(host_data(N))
        ALLOCATE(device_data(N))

        ! Initialize host data
        DO i = 1, N
          host_data(i) = REAL(i)
        END DO

        ! Map host data to device data
        CALL acc_map_data(host_data, device_data, N*4)

        ! Copy data from host to device
        CALL acc_memcpy_to_device(device_data, host_data, N*4)

        ! Check that data was correctly copied
        DO i = 1, N
          IF (device_data(i) .NE. REAL(i)) THEN
            errors = errors + 1
          END IF
        END DO

        ! Unmap data
        CALL acc_unmap_data(host_data)

        ! Deallocate host memory
        DEALLOCATE(host_data)
        DEALLOCATE(device_data)

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM

File: Deepseek-Coder-33b-Instruct/stage2/acc_map_data.cpp
#include <stdlib.h>
#include <openacc.h>
#include "acc_testsuite.h"

#ifndef T1
//T1:acc map data,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);

    const int size = 100;
    int *host_data = (int*)malloc(size * sizeof(int));
    int *device_data;

    // Initialize host data
    for(int i = 0; i < size; i++) {
        host_data[i] = i;
    }

    // Map host data to device
    #pragma acc data copyin(host_data[0:size])
    {
        device_data = (int*)acc_malloc(size * sizeof(int));
        #pragma acc update device(host_data[0:size]) async(1)
        #pragma acc parallel loop async(2)
        for(int i = 0; i < size; i++) {
            device_data[i] = host_data[i];
        }
        #pragma acc wait
    }

    // Check if data is correctly mapped
    for(int i = 0; i < size; i++) {
        if(device_data[i] != host_data[i]) {
            err++;
        }
    }

    free(host_data);
    acc_free(device_data);

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}